Crashes in `gmp-with-cxx` as a dependency of `gcc`:

```
$ nix build --no-link -f. stdenv --keep-going
error: builder for '/nix/store/nsk8abj7kbfh07234b2ry84kmawccc99-gmp-with-cxx-6.2.1.drv' failed with exit code 2;
       last 10 log lines:
       > make[5]: Leaving directory '/build/gmp-6.2.1/tests/cxx'
       > make[4]: *** [Makefile:1097: check-TESTS] Error 2
       > make[4]: Leaving directory '/build/gmp-6.2.1/tests/cxx'
       > make[3]: *** [Makefile:1315: check-am] Error 2
       > make[3]: Leaving directory '/build/gmp-6.2.1/tests/cxx'
       > make[2]: *** [Makefile:823: check-recursive] Error 1
       > make[2]: Leaving directory '/build/gmp-6.2.1/tests'
       > make[1]: *** [Makefile:997: check-recursive] Error 1
       > make[1]: Leaving directory '/build/gmp-6.2.1'
       > make: *** [Makefile:1289: check] Error 2
       For full logs, run 'nix log /nix/store/nsk8abj7kbfh07234b2ry84kmawccc99-gmp-with-cxx-6.2.1.drv'.
error: 1 dependencies of derivation '/nix/store/vz6xhj7s1dvqhn2kv59cvgwbb1czc0k9-isl-stage3-0.20.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8fvr0ija739lafp0x8lz8l0fyvk43qlx-mpfr-4.1.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/py2x1dvgkvdz7bic84xrr1kni50kd8z4-mpfr-stage3-4.1.1.drv' failed to build
error: 2 dependencies of derivation '/nix/store/rqq52r9w558sfk3nyhy6nbfcp12f3g9d-libmpc-stage3-1.2.1.drv' failed to build
```

Looking at the crash `libstdc++.so.6` is loaded from `bootstrapTools`'s `lib` directory (would probably use `bootstrapTools` `glibc`) while `glibc` used is already from `nixpkgs`:

```
   1831407:     calling init: /nix/store/ys9d0j18y9zfjx2fmmb2cjdivp2nv4z4-glibc-2.35-224/lib/ld-linux-x86-64.so.2
   1831407:     calling init: /nix/store/ys9d0j18y9zfjx2fmmb2cjdivp2nv4z4-glibc-2.35-224/lib/libc.so.6
   1831407:     calling init: /nix/store/ys9d0j18y9zfjx2fmmb2cjdivp2nv4z4-glibc-2.35-224/lib/libgcc_s.so.1
   1831407:     calling init: /nix/store/ys9d0j18y9zfjx2fmmb2cjdivp2nv4z4-glibc-2.35-224/lib/libm.so.6
   1831407:     calling init: /nix/store/p4s4jf7aq6v6z9iazll1aiqwb34aqxq9-bootstrap-tools/lib/libstdc++.so.6
```

Current `staging` does not seem broken. Cherry-picking b470a6b21293d0fa29d35f7e7ac180c5b0311e87
`linux/bootstrap-tools: move libstdc++ out of default library search path` alone does not make the crash go away.

I'll try to bisect the fix to see if it's intentional and rebase against `staging`.

--- a/pkgs/development/libraries/glibc/default.nix
+++ b/pkgs/development/libraries/glibc/default.nix
@@ -76,11 +76,16 @@ in
     # - clang-wrapper in cross-compilation
     # Last attempt: https://github.com/NixOS/nixpkgs/pull/36948
     preInstall = lib.optionalString (stdenv.hostPlatform == stdenv.buildPlatform) ''
-      if [ -f ${stdenv.cc.cc}/lib/libgcc_s.so.1 ]; then
+      if [ -f ${lib.getLib stdenv.cc.cc}/lib/libgcc_s.so.1 ]; then
           mkdir -p $out/lib
-          cp ${stdenv.cc.cc}/lib/libgcc_s.so.1 $out/lib/libgcc_s.so.1
+          cp ${lib.getLib stdenv.cc.cc}/lib/libgcc_s.so.1 $out/lib/libgcc_s.so.1
           # the .so It used to be a symlink, but now it is a script
-          cp -a ${stdenv.cc.cc}/lib/libgcc_s.so $out/lib/libgcc_s.so
+          cp -a ${lib.getLib stdenv.cc.cc}/lib/libgcc_s.so $out/lib/libgcc_s.so
+          # wipe out reference to previous libc it was built against
+          chmod +w $out/lib/libgcc_s.so.1
+          # rely on default RUNPATHs of the binary and other libraries
+          # Do no force-pull wrong glibc.
+          patchelf --remove-rpath $out/lib/libgcc_s.so.1
       fi
     '';
 

